File: cgrt/src/main/java/sron/cg/lang/IGraph.java
/*
 * Copyright 2016 Simón Oroño & La Universidad del Zulia
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package sron.cg.lang;

interface IGraph {
    int getSize();

    boolean containsVertex(int idx);

    boolean containsEdge(int source, int target);

    IGraph addVertex(int n);

    void addEdge(int source, int target);

    void removeEdge(int source, int target);

    void removeAllEdges();
}

File: cgrt/src/main/kotlin/sron/cg/runtime/graph/IGraph.kt
/*
 * Copyright 2016 Simón Oroño & La Universidad del Zulia
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package sron.cg.runtime.graph

interface IGraph {
    fun containsVertex(idx: Int): Boolean
    fun containsEdge(source: Int, target: Int): Boolean

    fun addVertex(n: Int): IGraph
    fun addEdge(source: Int, target: Int)

    fun removeEdge(source: Int, target: Int)
    fun removeAllEdges()
}
